hooks/VDM.js

```diff
@@ -262,18 +300,121 @@ module.exports = function (input, session) {
+    }
     values.push({
       path: 'sensors.ais.functionalId',
       value: data.fid,
     })
   }
 
+  if (data.lon && data.lat) {
+    values.push({
+      path: 'navigation.position',
+      value: {
+        longitude: data.lon,
+        latitude: data.lat,
+      },
+    })
+  }
+
+  [
+    ['avgwindspd', 'wind.averageSpeed', knotsToMs],
+    ['windgust', 'wind.gust', knotsToMs],
+    ['winddir', 'wind.directionTrue', degToRad],
+    ['windgustdir', 'wind.gustDirectionTrue', degToRad],
+    ['airtemp', 'outside.temperature', cToK],
+    ['relhumid', 'outside.relativeHumidity', (v) => v],
+    ['dewpoint', 'outside.dewPointTemperature', cToK],
+    ['airpress', 'outside.pressure', (v) => v * 100],
+    ['waterlevel', 'water.level', (v) => v],
+    ['signwavewhgt', 'water.waves.significantHeight', (v) => v],
+    ['waveperiod', 'water.waves.period', (v) => v],
+    ['wavedir', 'water.waves.directionTrue', degToRad],
+    ['swellhgt', 'water.swell.height', (v) => v],
+    ['swellperiod', 'water.swell.period', (v) => v],
+    ['swelldir', 'water.swell.directionTrue', degToRad],
+    ['watertemp', 'water.temperature', cToK],
+    ['salinity', 'water.salinity', (v) => v],
+  ].forEach(([propName, path, f]) => {
+    if (data[propName] !== undefined) {
+      contextPrefix = 'meteo.'
+      values.push({
+        path: `environment.observation.` + path,
+        value: f(data[propName]),
+      })
+    }
+  });
+
+  [
+    ['ice', 'water.ice', iceTable],
+    ['precipitation', 'outside.precipitation', precipitationType],
+    ['seastate', 'water.seastate', beaufortScale],
+    ['waterlevelten', 'water.levelTendency', statusTable],
+    ['airpressten', 'outside.pressureTendency', statusTable],
+  ].forEach(([propName, path, f]) => {
+    if (data[propName] !== undefined) {
+      contextPrefix = 'meteo.'
+        const comment = f[data[propName]]
+        const value = data[propName]
+      values.push({
+        path: `environment.observation.` + path,
+        value: {
+          value,
+          comment,
+        }
+      })
+    }
+  });
+
+  if (data.surfcurrspd !== undefined || data.surfcurrdir !== undefined) {
+    contextPrefix = 'meteo.'
+    const drift = utils.transform(data.surfcurrspd, 'knots', 'ms')
+    const set = utils.transform(data.surfcurrdir, 'deg', 'rad')
+    values.push({
+      path: 'environment.observations.current',
+      value: {
+        set,
+        drift,
+      },
+    })
+  }
+
+  if (data.horvisib !== undefined && data.horvisibrange !== undefined) {
+    contextPrefix = 'meteo.'
+    const value = utils.transform(data.horvisib, 'nm', 'm')
+    const comment = data.horvisibrange
+    values.push({
+      path: 'environment.observations.horizontalVisibility',
+      value: {
+        value,
+        comment,
+      },
+    })
+  }
+
+  if (data.utcday !== undefined && data.utchour !== undefined && data.utcminute !== undefined) {
+    contextPrefix = 'meteo.'
+    const utcday = data.utcday
```

Comment:
Usually a ISOTimeString is used to represent timestamps.
Is there a reason this is being broken out?

Reply:
This was requested by @Hakansv.
As it is partial date data and not usable as it is.
I would hear @Hakansv use case for this. Minimum resolution is 1min so don't know does this add any additional info SK timestamp which is received data timestap instead of generated by source.